### PR TITLE
Closing connections enhancements

### PR DESCRIPTION
**Pull request by Flood1993**
Various improvements on how to close a connection (socket).

### CODE REVIEW

**Comment by Flood1993** [comment]
I will squash commits if this gets accepted. Since there is not much code, I think it is easier to review all changes directly instead of per commit @Peque .

File: docs/source/considerations.rst [hunk 1]
```diff
@@ -26,6 +26,32 @@ For more information simply refer to the
 `ØMQ guide <http://zguide.zeromq.org/page:all>`_.
 
 
+Closing connections
+===================
+
+For closing a specific connection from an agent we need to call the
+``close_socket()`` method, which takes the alias of the socket from the
```
Comment (Peque) [inline docs/source/considerations.rst]: Maybe rename `close_socket` to `close`? (bind, connect, close; all are referring to sockets)

File: docs/source/considerations.rst [hunk 2]
```diff
@@ -26,6 +26,32 @@ For more information simply refer to the
...
+connection we want to close as a parameter.
+
+.. code-block:: python
+
+   agent.bind('PUB', alias='connection')
+   ...
+   agent.close_socket('connection')
+
+There is also a ``close_sockets()`` method, which takes no parameters and
```
Comment (Peque) [inline docs/source/considerations.rst]: And maybe `close_all` here as well? (for consistency)

File: osbrain/agent.py [hunk 3]
```diff
@@ -1511,13 +1511,48 @@ def get_unique_external_zmq_sockets(self):
...
+        for k, v in self.socket.items():
+            if v == sock:
+                entries_to_delete.append(k)
+
+        for entry in entries_to_delete:
+            del self.socket[entry]
+
+        sock.close(linger=get_linger())
+
     def close_sockets(self):
```
Comment (Peque) [inline osbrain/agent.py]: And maybe `close_all` here as well? (for consistency)

File: osbrain/agent.py [hunk 4]
```diff
@@ -1511,13 +1511,48 @@ def get_unique_external_zmq_sockets(self):
 
         return set(external_sockets)
 
+    def has_socket(self, alias):
+        """
+        Return whether the agent has the passed socket internally stored.
+        """
+        return alias in self.socket
+
+    def close_socket(self, alias):
```
Comment (Peque) [inline osbrain/agent.py]: Maybe `close` instead of `close_socket`?

**Pull request merged**